include/deal.II/grid/manifold.h

```diff
@@ -610,12 +615,12 @@ class FlatManifold : public Manifold<dim, spacedim>
    * A periodicity 0 along one direction means no periodicity. This is the
    * default value for all directions.
    */
-  const Point<spacedim> periodicity;
+  const Tensor<1,spacedim> periodicity;
 
-  DeclException4(ExcPeriodicBox, int, Point<spacedim>, Point<spacedim>, double,
+  DeclException4(ExcPeriodicBox, int, Point<spacedim>, double, double,
```

Comment:
I find [-arg4,arg3] somewhat confusing. Wouldn't it be enough to display ``[0,arg3]`` and not pass in arg4?